symphony-bdk-core/src/test/java/com/symphony/bdk/core/auth/AuthenticatorFactoryTest.java

```diff
@@ -123,6 +124,45 @@ void testGetOboCertificateAuthenticator() throws AuthInitializationException {
     assertEquals(OboAuthenticatorCertImpl.class, oboAuthenticator.getClass());
   }
 
+  @Test
+  void testGetExtAppAuthenticatorWithValidPrivateKey(@TempDir Path tempDir) throws AuthInitializationException {
+
+    final BdkConfig config = createRsaConfig(() -> {
+      final Path privateKeyPath = tempDir.resolve(UUID.randomUUID().toString() + "-privateKey.pem");
+      writeContentToPath(RSA_PRIVATE_KEY, privateKeyPath);
+      return privateKeyPath.toAbsolutePath().toString();
+    });
+
+    final AuthenticatorFactory factory = new AuthenticatorFactory(config, DUMMY_API_CLIENT_FACTORY);
+    final ExtensionAppAuthenticator botAuth = factory.getExtensionAppAuthenticator();
+    assertNotNull(botAuth);
```

Comment:
I would remove the assertNotNull for brevity as we try to access it in the assertEquals